test/server/spec/bin/builder-init.spec.js

```diff
@@ -0,0 +1,411 @@
+"use strict";
+
+/**
+ * These are _almost_ functional tests as we're basically invoking the entire
+ * application, just:
+ *
+ * - Mocking filesystem
+ * - Stubbing stdin to return canned responses to prompts
+ */
+var childProc = require("child_process");
+var crypto = require("crypto");
+var zlib = require("zlib");
+var stream = require("stream");
+var temp = require("temp").track();
+var _ = require("lodash");
+var Prompt = require("inquirer/lib/prompts/base");
+var _eval = require("eval");
+
+var run = require("../../../../bin/builder-init");
+var Task = require("../../../../lib/task");
+var pkg = require("../../../../package.json");
+
+var base = require("../base.spec");
+
+// Helpers
+// **Note**: It would be great to just stub stderr, stdout in beforeEach,
+// but then we don't get test output. So, we manually stub with this wrapper.
+var stdioWrap = function (fn) {
+  return function (done) {
+    base.sandbox.stub(process.stdout, "write");
+
+    var _done = function (err) {
+      process.stdout.write.restore();
+      done(err);
+    };
+
+    try {
+      return fn(_done);
+    } catch (err) {
+      return _done(err);
+    }
+  };
+};
+
+// Mock key I/O parts of the flow.
+/*eslint-disable max-statements*/
+var mockFlow = function (extracted, root) {
```

Comment:
I know!

The cool thing is that `mock-fs` has been wonderful for having a ton of "file-based" tests that don't actually touch any filesystem so that I can just define them with objects in place in the tests that consume them.